core/src/main/java/hudson/PluginManager.java

```diff
@@ -1350,6 +1350,55 @@ public static boolean isNonMetaLabel(String label) {
         return !("adopt-this-plugin".equals(label) || "deprecated".equals(label));
     }
 
+    /**
+     Returns the sort order index for the given {@link PluginWrapper} on {@code installed.jelly}.
+     <ol>
+     <li>Plugins marked for deletion (temporary until restart)</li>
+     <li>Inactive and disabled plugins that can be enabled (often present for a while before uninstalled)</li>
+     <li>Inactive and disabled plugins that cannot be enabled (often present for a while before uninstalled)</li>
+     <li>Active plugins that have been disabled (temporary until restart)</li>
+     <li>Inactive plugins that have just been enabled (temporary until restart)</li>
+     <li>Enabled and active plugins that can be freely disabled/uninstalled from a dependency POV</li>
+     <li>Enabled and active plugins with implied dependants, can be disabled but might be unsafe</li>
+     <li>Enabled and active plugins with mandatory dependants, cannot be disabled.</li>
+     </ol>
+     <p>
+     This groups by visual presentation, with the states that can be more readily interacted with towards the top, pushing required dependencies to the bottom.
+     '1' / '2-4' / '5-8' is trivially reflected on the UI through the state/existence of the toggle button.
+     </p>
+     */
+    @Restricted(NoExternalUse.class) // Jelly only
+    public int getSortIndex(PluginWrapper p) {
+        if (p.isDeleted()) {
+            return 1; // temporary status
+        }
+        if (!p.isEnabled()) {
+            if (p.isActive()) {
+                return 4; // temporary status
+            }
+            // TODO: Disabled but cannot be enabled
+            for (Dependency d : p.getMandatoryDependencies()) {
+                PluginWrapper dependency = getPlugin(d.shortName))
```

Comment:
```suggestion
                PluginWrapper dependency = getPlugin(d.shortName);
```
Let's make it build 😎